lib/type-registry.js

```diff
@@ -0,0 +1,118 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: strong-remoting
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+var ArrayConverter = require('./types/array');
+var assert = require('assert');
+var debug = require('debug')('strong-remoting:types');
+var g = require('strong-globalize')();
+
+module.exports = TypeRegistry;
+
+function TypeRegistry(options) {
+  this._options = options || {};
+  this._types = Object.create(null);
+  this._registerBuiltinTypes();
+}
+
+TypeRegistry.prototype.registerType = function(typeName, converter) {
+  assert(typeof typeName === 'string' && typeName,
+    'typeName must be a non-empty string');
+  assert(typeof converter === 'object' && converter,
+    'converter must be an object');
+  assert(typeof converter.fromTypedValue === 'function',
+    'converter.fromTypedValue must be a function');
+  assert(typeof converter.fromSloppyValue === 'function',
+    'converter.fromSloppyValue must be a function');
+  assert(typeof converter.validate === 'function',
+    'converter.validate must be a function');
+
+  typeName = typeName.toLowerCase();
+
+  if (typeName in this._types) {
+    if (this._options.warnWhenOverridingType !== false)
+      g.warn('Warning: overriding remoting type %s', typeName);
+    else
+      debug('Warning: overriding remoting type %s', typeName);
```

Comment:
I don't think we have a rule for that (yet). Since `eslint` checks are passing, the style should be good.

TBH, I don't really mind. If you think I should add curly braces, then I'll do so. Please let me know.